sonic-xcvrd/xcvrd/xcvrd_utilities/media_settings_parser.py

```diff
@@ -0,0 +1,317 @@
+import json
+import os
+import ast
+
+from sonic_py_common import device_info, logger
+from swsscommon import swsscommon
+from xcvrd import xcvrd
+
+g_dict = {}
+
+LANE_SPEED_KEY_PREFIX = "speed:"
+VENDOR_KEY = 'vendor_key'
+MEDIA_KEY = 'media_key'
+LANE_SPEED_KEY = 'lane_speed_key'
+SYSLOG_IDENTIFIER = "xcvrd"
+helper_logger = logger.Logger(SYSLOG_IDENTIFIER)
+
+
```

Comment:
@tshalvi add a comment, this parser parses the NPU or ASIC side SerDes custom SI settings.